File: Sesion 8/Actividades/Actividad 1/actividad1_2.py
import math

# Declaración de la clase abstracta Figura
class Figura:
    # Método virtual puro para calcular el área
    def calcularArea(self):
        pass

# Clase derivada Círculo
class Circulo(Figura):
    def __init__(self, r):
        self.radio = r

    def calcularArea(self):
        return math.pi * self.radio * self.radio

# Clase derivada Rectángulo
class Rectangulo(Figura):
    def __init__(self, b, h):
        self.base = b
        self.altura = h

    def calcularArea(self):
        return self.base * self.altura

# Clase derivada Triángulo
class Triangulo(Figura):
    def __init__(self, b, h):
        self.base = b
        self.altura = h

    def calcularArea(self):
        return (self.base * self.altura) / 2

# Ejemplo de uso
circulo = Circulo(5)
rectangulo = Rectangulo(4, 6)
triangulo = Triangulo(3, 8)

print("Área del círculo:", circulo.calcularArea())
print("Área del rectángulo:", rectangulo.calcularArea())
print("Área del triángulo:", triangulo.calcularArea())

File: Sesion 8/Actividades/Actividad 1/actividad1_2.java
// Declaración de la clase abstracta Figura
abstract class Figura {
    // Método abstracto para calcular el área
    abstract float calcularArea();
}

// Clase derivada Círculo
class Circulo extends Figura {
    private float radio;

    Circulo(float r) {
        radio = r;
    }

    float calcularArea() {
        return (float) (Math.PI * radio * radio);
    }
}

// Clase derivada Rectángulo
class Rectangulo extends Figura {
    private float base, altura;

    Rectangulo(float b, float h) {
        base = b;
        altura = h;
    }

    float calcularArea() {
        return base * altura;
    }
}

// Clase derivada Triángulo
class Triangulo extends Figura {
    private float base, altura;

    Triangulo(float b, float h) {
        base = b;
        altura = h;
    }

    float calcularArea() {
        return (base * altura) / 2;
    }
}

public class Main {
    public static void main(String[] args) {
        Circulo circulo = new Circulo(5);
        Rectangulo rectangulo = new Rectangulo(4, 6);
        Triangulo triangulo = new Triangulo(3, 8);

        System.out.println("Área del círculo: " + circulo.calcularArea());
        System.out.println("Área del rectángulo: " + rectangulo.calcularArea());
        System.out.println("Área del triángulo: " + triangulo.calcularArea());
    }
}
